webapp/cas/config.go

```
@@ -0,0 +1,141 @@
+}
+
+func parseTicketEncoding(raw string) (TicketDecoder, error) {
+	switch raw {
+	case "base64":
+		return base64.StdEncoding.DecodeString, nil
+	default:
+		return nil, errors.New("unable to find TicketEncoding")
+	}
+}
+
+func parseChallengeHash(raw string) (ChallengeHasher, error) {
+	switch raw {
+	case "sha256":
+		return func(data []byte) []byte {
+			ret := sha256.Sum256(data)
+			return ret[:]
+		}, nil
+	default:
+		return nil, errors.New("unable to find ChallengeHash")
+	}
+}
+
+func ParseConfig(tomlRaw []byte) (*Config, error) {
+	var tomlConf struct {
+		ByzCoinID string
+		Servers   []struct {
+			Address       network.Address
+			Suite, Public string
+		}
+
+		ServiceToCoinInstanceIDs map[string]string
+
+		TicketEncoding     string
+		ChallengeSize      uint
+		ChallengeHash      string
+		TxArgumentName     string
+		TxValidityDuration string
+	}
+	if err := toml.Unmarshal(tomlRaw, &tomlConf); err != nil {
+		return nil, err
+	}
+
+	byzCoinID, err := parseSkipBlockID(tomlConf.ByzCoinID)
+	if err != nil {
+		return nil, err
+	}
+
+	servers := make([]*network.ServerIdentity, len(tomlConf.Servers))
+	for i, s := range tomlConf.Servers {
+		suite, err := suites.Find(s.Suite)
+		if err != nil {
+			return nil, err
+		}
+
+		point, err := encoding.StringHexToPoint(suite, s.Public)
+		if err != nil {
+			return nil, err
+		}
+
+		servers[i] = network.NewServerIdentity(point, s.Address)
+	}
+
+	ticketDecoder, err := parseTicketEncoding(tomlConf.TicketEncoding)
+	if err != nil {
+		return nil, err
+	}
+
+	challengeHash, err := parseChallengeHash(tomlConf.ChallengeHash)
+	if err != nil {
+		return nil, err
+	}
+
+	serviceToCoinInstanceIDs := make(map[string]skipchain.SkipBlockID)
+	for k, v := range tomlConf.ServiceToCoinInstanceIDs {
+		serviceToCoinInstanceIDs[k], err = hex.DecodeString(v)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	txValidityDuration, err := time.ParseDuration(tomlConf.TxValidityDuration)
+
+	return &Config{
+		byzCoinID,
```

I think `golint` will complain here that you don't name the fields. And if it doesn't complain, I will ;) I got used to it, because, even if most of the variables correspond to the field names, not all do. So it's nice to have the field name with it.

---

(it's not complaining :p) It's not the same meaning, using named composite literals set the zero values for missing keys, ie doesn't msg/warn/fail when adding new fields (silent fail IMHO).

From the [Effective go documentation](https://golang.org/doc/effective_go.html#composite_literals)
> The fields of a composite literal are laid out in order and must all be present. However, by labeling the elements explicitly as field:value pairs, the initializers can appear in any order, with the missing ones left as their respective zero values.